sormas-ui/src/main/java/de/symeda/sormas/ui/events/eventParticipantMerge/EventParticipantPickLeadEventParticipant.java

```diff
@@ -0,0 +1,244 @@
+package de.symeda.sormas.ui.events.eventParticipantMerge;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.vaadin.server.ExternalResource;
+import com.vaadin.ui.Component;
+import com.vaadin.ui.HorizontalLayout;
+import com.vaadin.ui.Label;
+import com.vaadin.ui.Link;
+import com.vaadin.ui.VerticalLayout;
+import com.vaadin.v7.ui.CustomField;
+
+import de.symeda.sormas.api.FacadeProvider;
+import de.symeda.sormas.api.event.EventDto;
+import de.symeda.sormas.api.event.EventParticipantDto;
+import de.symeda.sormas.api.event.EventParticipantSelectionDto;
+import de.symeda.sormas.api.event.EventSelectionDto;
+import de.symeda.sormas.api.i18n.Captions;
+import de.symeda.sormas.api.i18n.I18nProperties;
+import de.symeda.sormas.api.i18n.Strings;
+import de.symeda.sormas.api.infrastructure.district.DistrictDto;
+import de.symeda.sormas.api.person.PersonHelper;
+import de.symeda.sormas.api.utils.DataHelper;
+import de.symeda.sormas.ui.SormasUI;
+import de.symeda.sormas.ui.events.EventDataView;
+import de.symeda.sormas.ui.utils.CssStyles;
+import de.symeda.sormas.ui.utils.VaadinUiUtil;
+
+public class EventParticipantPickLeadEventParticipant extends CustomField<List<String>> {
+
+	protected List<EventParticipantDto> eventParticipantDtos;
+	private Map<String, EventParticipantDuplicate> eventParticipantSelectionDtos;
+	private Map<String, EventParticipantDuplicateGrid> eventParticipantMergeSelectionGrid = new HashMap<>();
```

Comment:
does this need to be a Map?

please give a better name

Reply:
well, the key contains the eventUuid. In case we have an error in which the grid contains 2 event participants from 2 different events it will be harder to debug without the key. 
renamed